commons/c2cgeoportal_commons/models/main.py

```diff
@@ -311,7 +311,7 @@ class LayergroupTreeitem(Base):
         backref=backref(
             'children_relation',
             order_by='LayergroupTreeitem.ordering',
-            cascade='save-update,merge,delete,delete-orphan',
+            cascade='save-update,merge,delete,delete-orphan,expunge',
```

Comment:
We use expunge in duplicate functionnality (in case or a layer_group we duplicate the model instance with children), just to populate a form to propose to the user, but we do not want to add it to the session. But as we duplicate relations, the duplicated instance is added to the session.
So we have put this block in a : with no_autoflush

Reply:
https://github.com/camptocamp/c2cgeoform/blob/master/c2cgeoform/views/abstract_views.py#L302:L

and we want to expunge duplicated instances from session in this no_autoflush block.

concerning the parents, we do not need this behavior.
we want to expunge children, metatadatas, dimension values (instances that are created through the form of a parent object) that should expunge with that parent object. But we do not want parent to be expunged with the child.

Reply:
I'm not sure to understand every thing, I hope that's correct :-)